website/docs/icons/library/index.js

```diff
@@ -71,36 +79,64 @@ export default class Index extends Component {
       );
     }
 
-    // alphabetize the filtered icons by category and then by name
-    filteredIcons
-      .sort((a, b) => {
-        return a.category.localeCompare(b.category);
-      })
-      .sort((a, b) => {
-        if (a.category === b.category) {
-          a.iconName.localeCompare(b.iconName);
-        }
+    // 2) Sort icons by Group Type:
+    // Sort alphabetically by iconName:
+    if (this.selectedGroupType === 'alphabetical') {
+      filteredIcons.sort((a, b) => {
+        return a.iconName.localeCompare(b.iconName);
+      });
 
-        return 0;
+      filteredIcons.forEach((icon) => {
+        const category = '';
+        if (!filteredGroupedIcons[category]) {
+          filteredGroupedIcons[category] = [];
+        }
+        filteredGroupedIcons[category].push(icon);
       });
 
-    // Group all filtered icons by category
-    filteredIcons.forEach((icon) => {
-      const category = icon.category;
-      if (!filteredGroupedIcons[category]) {
-        filteredGroupedIcons[category] = [];
-      }
-      filteredGroupedIcons[category].push(icon);
-    });
+      // Sort icons by category then iconName:
+    } else if (this.selectedGroupType === 'category') {
+      filteredIcons
+        .sort((a, b) => {
+          return a.category.localeCompare(b.category);
+        })
+        .sort((a, b) => {
+          if (a.category === b.category) {
+            a.iconName.localeCompare(b.iconName);
+          }
+          return 0;
+        });
+      // Group all filtered icons by category
+      filteredIcons.forEach((icon) => {
+        const category = icon.category;
+        if (!filteredGroupedIcons[category]) {
+          filteredGroupedIcons[category] = [];
+        }
+        filteredGroupedIcons[category].push(icon);
+      });
```

Comment:
Currently I'm repeating this logic which should probably be unified somehow.

Reply:
Yes, if you can review part of this logic to make it easier to follow, that would be great. But it's not a blocker, so I'll let you decide the best course of action.

Reply:
Thanks, I cleaned it up now.